dialer.go

```diff
@@ -159,7 +163,12 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 		dialerID:       uuid.New().String(),
 		iamTokenSource: cfg.tokenSource,
 		dialFunc:       cfg.dialFunc,
+		metrics:        mc,
+		openConns:      make(map[string]int64),
+		done:           make(chan struct{}),
 	}
+	// TODO: don't start the goroutine unless a user has enabled metrics.
+	go d.reportOpenConns(ctx)
```

Comment:
This seems like it'll use the context that constructed the dialer for background work. Are we use we want to do that?

Reply:
I've just dropped this.